pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -40,21 +40,39 @@ var _ volume.VolumeHost = &PersistentVolumeRecycler{}
 // This controller will Recycle those volumes whose reclaim policy is set to PersistentVolumeReclaimRecycle and make them
 // available again for a new claim.
 type PersistentVolumeRecycler struct {
-	volumeController *framework.Controller
-	stopChannel      chan struct{}
-	client           recyclerClient
-	kubeClient       client.Interface
-	pluginMgr        volume.VolumePluginMgr
-	cloud            cloudprovider.Interface
+	volumeController  *framework.Controller
+	stopChannel       chan struct{}
+	client            recyclerClient
+	kubeClient        client.Interface
+	pluginMgr         volume.VolumePluginMgr
+	cloud             cloudprovider.Interface
+	maximumRetryCount int
+	syncPeriod        time.Duration
+	// Local cache of failed recycle / delete operations. Map volume.Name -> status of the volume.
+	// Only PVs in Released state have an entry here.
+	releasedVolumes map[string]releasedVolumeStatus
```

Comment:
Hmm.. I am not too comfortable with this approach. Wouldn't the volume state be better stored as an annotation on the PV ? You should be able to look in one place and figure out the sate of a PV. This makes it easier people other than the original author to debug volume problems. In addition if we end up using the State design pattern to clean volume states up we'll have to undo this and put it back in the volume.